### PR TITLE
fix: retry on docker inpsect timeouts

### LINKED ISSUE #453: Gracefully handle `docker inspect` timeout
We periodically see the docker service get stuck responding to `docker inspect` here:

https://github.com/opensafely-core/job-runner/blob/master/jobrunner/executors/local.py#L188

This command timesout after the default of 300s, and then fails the job with internal error.

This is a good proportion (maybe around 50%?) of "Internal Errors" that the user and tech-support see, and it shouldn't be. It also means the main run loop of job-server is blocked for 300s, which is also bad. 

This particular `inspect` call is something we do to every job on every loop, and failing with a timeout it is not fatal (unlike say failing to get outputs, or find the container).

I think we should:

1) set a much shorted timeout, say 30s 
2) If it times out, *log it*, but raise an error that will not result in a job failure with internal error, but rather, the job will be skipped this run of the loop, and retried on the next run (as per normal for non-fatal errors).

This should allow us to ride out when docker has a hitch without failing a bunch of jobs.

However, this could result in a docker hang up getting jobs stuck. But such a case seems fairly catastrophic anyway, and would presumably result in many other failures. We could potentially handle this (e.g. tracking the number of timeouts and hard erroring if they reach a threshold, or similar schemes), but I'd be tempted to try it first.

### PR DESCRIPTION
**Pull request by bloodearnest**
Fixes #453

This adds a tuneable config for how many retries we allow before
failure.

### CODE REVIEW

File: jobrunner/executors/local.py [hunk 1]
```diff
@@ -187,9 +188,15 @@ def cleanup(self, job):
...
         name = container_name(job)
-        job_running = docker.container_inspect(
-            name, "State.Running", none_if_not_exists=True
-        )
+        try:
+            job_running = docker.container_inspect(
+                name,
+                "State.Running",
+                none_if_not_exists=True,
+                timeout=10,
```
Comment (lucyb) [inline jobrunner/executors/local.py:196]: Why 10?

File: jobrunner/run.py [hunk 2]
```diff
@@ -188,7 +196,19 @@ def handle_job(job, api, mode=None, paused=None):
         )
         return
 
-    initial_status = api.get_status(definition)
+    try:
```
Comment (lucyb) [inline jobrunner/run.py:199]: Not one for this change, but `handle_job` is getting quite long. It took a while for me to read through and work out what was happening here. Would be worth thinking about how this function could be refactored in future.

**Review by lucyb — APPROVED** [review]
Looks good.

**Review by evansd — APPROVED** [review]
+1 I think this will be a big win in terms of operational stability under load. Thanks for taking it on.

File: jobrunner/run.py (hunk 2 above)
Comment (bloodearnest) [reply jobrunner/run.py:199]: That's exactly what my re-thinking scheduling doc suggests :)

Basically, split the logic between two loops/functions, one that handles PENDING and one that handles RUNNING.

Comment (bloodearnest) [reply jobrunner/run.py:199]: But also, large functions are often unavoidable in schedulers, IME, as its essentially a state transition engine, and those often don't decompose well.

**Pull request merged**